From 07874570e76218336934a58264b39224ed17485c Mon Sep 17 00:00:00 2001
From: Jiping Ma <jiping.ma2@windriver.com>
Date: Sun, 12 Jun 2022 23:54:50 -0700
Subject: [PATCH 1/3] kdump-tools: add vmlinuz and initrd.img soft link.

Default vmlinuz is in the folder /boot, but our vmlinuz is in /boot/1.
so create the soft link for vmlinuz and initrd.img.

Signed-off-by: Jiping Ma <jiping.ma2@windriver.com>
Signed-off-by: Catalin Rapiteanu <viorel-catalin.rapiteanu@windriver.com>
---
 debian/kdump-config.in | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/debian/kdump-config.in b/debian/kdump-config.in
index 8aa91b6..8ac8239 100755
--- a/debian/kdump-config.in
+++ b/debian/kdump-config.in
@@ -675,6 +675,12 @@ kdump_create_symlinks()
 		exit 1
 	fi
 
+	if [ ! -e "/boot/vmlinux-${kernel_version}" ] && [ ! -e "/boot/vmlinuz-${kernel_version}" ]; then
+		ln -s /boot/1/vmlinuz-${kernel_version} /boot/vmlinuz-${kernel_version}
+	fi
+	if [ ! -f "${KDUMP_DIR}/initrd.img-${kernel_version}" ]; then
+		ln -s /boot/1/initramfs /var/lib/kdump/initrd.img-${kernel_version}
+	fi
 	if [ -e "/boot/vmlinux-${kernel_version}" ] || [ -e "/boot/vmlinuz-${kernel_version}" ]; then
 		create_symlink vmlinuz "$kernel_version"
 
-- 
2.47.3

